File: 2024/01B.py
import sys

def solve(input):
    nums1 = [int(line.split()[0]) for line in input.splitlines()]
    nums2 = [int(line.split()[1]) for line in input.splitlines()]

    result = 0
    for num1 in nums1:
        result += num1 * nums2.count(num1)

    return result
        
if __name__ == '__main__':
    print(solve(open(sys.argv[1]).read()))  

File: 2024/01B.js


function solve(input) {
  const l1 = input.split('\n').map((row) => parseInt(row.split('  ')[0]));
  const l2 = input.split('\n').map((row) => parseInt(row.split('  ')[1]));

  var result = 0;
  for (let i = 0; i < l1.length ; i++){
    result += l1[i] * l2.filter((v) => v == l1[i]).length;
  }

  return result;
}

console.log(solve(require('node:fs').readFileSync(process.argv[2], 'utf8').trim()));